app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,93 @@
+class OrdersController < ApplicationController
+  #look up callbacks
+  #also, in order to make filter work, buttons to do stuff to orders need to send an id into params. (see guest_authorize method, which uses id)
+  #this is to help make it so only the person who started an order can do stuff to it.
+  before_action :guest_authorize, :only =>[:clear_cart, :cancel_as_guest, :pay]
+  before_action :navbar_categories, only: [:cart]
+
+
+  #the order from the perspective of the merchant
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  #the order from the perspective of the guest
+  #stuff needed to view the cart page
+  def cart
+    if !session[:order_id]
+      @cart_status = "empty"
+    end
+  end
+
+  #before you pay, clearing cart destroys order and orderitems.
+  #after you've paid, you instead cancel the order, and the order sticks around in the database
+  def clear_cart
+    @order = Order.find(session[:order_id])
+    @order.destroy!
+    session[:order_id] = nil
+    redirect_to root_path
+  end
+
+  def checkout
+    @order = Order.find(session[:order_id])
```

Comment:
To DRY out this code some more, defining the order can be put into it's own method and then called in a `before_action`. For example:
```ruby
before_action :find_order, except: [:cart]

def find_order
  @order = Order.find(session[:order_id])
end
```